usr/share/rear/prep/NETFS/default/05_check_NETFS_requirements.sh

```diff
@@ -6,49 +6,48 @@
 # example: file:///path
 # example: iso://backup/
 # example: sshfs://user@host/G/rear/
+# example: ftpfs://user@host/rear/
 
 [[ "$BACKUP_URL" || "$BACKUP_MOUNTCMD" ]]
+# FIXME: The above test does not match the error message below.
+# To match the the error message the test should be
+# [[ "$BACKUP_URL" || ( "$BACKUP_MOUNTCMD" && "$BACKUP_UMOUNTCMD" ) ]]
+# but I <jsmeix@suse.de> cannot decide if there is a subtle reason for the omission.
 StopIfError "You must specify either BACKUP_URL or BACKUP_MOUNTCMD and BACKUP_UMOUNTCMD !"
 
 if [[ "$BACKUP_URL" ]] ; then
-    local host=$(url_host $BACKUP_URL)
-    local scheme=$(url_scheme $BACKUP_URL)
-    local path=$(url_path $BACKUP_URL)
+    local scheme=$( url_scheme $BACKUP_URL )
+    local hostname=$( url_hostname $BACKUP_URL )
+    local path=$( url_path $BACKUP_URL )
 
     ### check for vaild BACKUP_URL schemes
     ### see https://github.com/rear/rear/issues/842
     case $scheme in
-        (nfs|cifs|usb|tape|file|iso|sshfs)
-          # do nothing for vaild BACKUP_URL schemes
-          :
-          ;;
+        (nfs|cifs|usb|tape|file|iso|sshfs|ftpfs)
+            # do nothing for vaild BACKUP_URL schemes
+            :
+            ;;
         (*)
-          Error "Invalid scheme '$scheme' in BACKUP_URL '$BACKUP_URL' (only nfs cifs usb tape file iso sshfs are valid)"
-          ;;
+            Error "Invalid scheme '$scheme' in BACKUP_URL '$BACKUP_URL' valid schemes: nfs cifs usb tape file iso sshfs ftpfs"
+            ;;
     esac
 
     ### set other variables from BACKUP_URL
-    case $scheme in
-        (usb)
-            if [[ -z "$USB_DEVICE" ]] ; then
-                USB_DEVICE="$path"
-            fi
-            ;;
-	(sshfs)
-	    # check if $host contains a '@' because then we use user@host format
-	    echo $host | grep -q '@' && {
-		sshfs_user="${host%%@*}"	# save the user
-		host="${host#*@}"		# remove user@
-		}
-	    ;;
-    esac
+    if [[ "usb" = "$scheme" ]] ; then
+        # if USB_DEVICE is not explicitly specified it is the path from BACKUP_URL
+        [[ -z "$USB_DEVICE" ]] && USB_DEVICE="$path"
+    fi
 
     ### check if host is reachable
-    if [[ "$PING" && "$host" ]] ; then
-        ping -c 2 "$host" >&8
-        StopIfError "Backup host [$host] not reachable."
+    if [[ "$PING" && "$hostname" ]] ; then
+        # TODO: it is questionable if it is a fatal error when it does not respond to a 'ping'
+        # because sometimes hosts are accessible via certain ports but do not respond to a 'ping'
```

Comment:
@jsmeix I agree with you, but due to lack of a better test, we can decide to drop it or leave it?
@schlomo what is your opinion?